tests/backup/backup_psa_test.go

```diff
@@ -1,30 +1,596 @@
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Validation of the manual backup [%s]", backupName))
+					backupNamesAllNs = append(backupNames, backupName)
+				}
+			})
+
+			Step("Default restore of applications by replacing the existing resources with NS level PSA and no cluster level setting", func() {
+				log.InfoD("Default restore of applications by replacing the existing resources with NS level PSA and no cluster level setting")
+				ctx, err := backup.GetAdminCtxFromSecret()
+				log.FailOnError(err, "Fetching px-central-admin ctx")
+
+				for _, backupName := range backupNamesAllNs {
+					defaultRestoreName := fmt.Sprintf("%s-%s", RestoreNamePrefix, backupName)
+					err = CreateRestoreWithReplacePolicyWithValidation(defaultRestoreName, backupName, make(map[string]string), SourceClusterName, BackupOrgID, ctx, make(map[string]string), ReplacePolicyDelete, scheduledAppContexts)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creating default restore for manual backup with replace policy [%s]", defaultRestoreName))
+					restoreNames = append(restoreNames, defaultRestoreName)
+				}
+			})
+
+			Step("Restoring only PVC from backup", func() {
+				ctx, err := backup.GetAdminCtxFromSecret()
+				log.FailOnError(err, "Fetching px-central-admin ctx")
+				log.InfoD("Restoring backed up applications")
+				for i, backupName := range backupNamesAllNs {
+					restoreContextMap := make(map[string][]*scheduler.Context)
+					appContextsToBackup := make([]*scheduler.Context, 0)
+					namespaceMapping := make(map[string]string)
+					restoreName := fmt.Sprintf("%s-%s-%v", RestoreNamePrefix, backupName, time.Now().Unix())
+					namespaceMapping[bkpNamespaces[i]] = bkpNamespaces[i] + "-restored-only-pvc-3"
+					_, err := CreateCustomRestoreWithPVCs(restoreName, backupName, namespaceMapping, SourceClusterName, BackupOrgID, ctx, make(map[string]string), bkpNamespaces[i])
+					log.FailOnError(err, "Restoring of backup [%s] has failed with name [%s] in namespace [%s]", backupName, restoreName, fmt.Sprintf(bkpNamespaces[i]+"-restored-only-pvc"))
+					appContextsToBackup = FilterAppContextsByNamespace(scheduledAppContexts, []string{bkpNamespaces[i] + "-restored-only-pvc-3"})
+					restoreContextMap[restoreName] = appContextsToBackup
+					err = ValidateCustomResourceRestores(ctx, BackupOrgID, []string{"PersistentVolumeClaim"}, restoreContextMap, SourceClusterName)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Validating custom resource restore - %v", restoreNames))
+					restoreNames = append(restoreNames, restoreName)
+				}
+			})
+
+		}
+
+	})
 	JustAfterEach(func() {
-		log.InfoD("Nothing to be deleted")
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+
+		// Make sure to revert the cluster level PSA settings
+		defer func() {
+			if psaFlag {
+				err := SetDestinationKubeConfig()
+				dash.VerifyFatal(err, nil, "Setting destination kubeconfig")
+				err = RevertClusterLevelPSA()
```

Comment:
If there is an error here, then it wont switch back to source cluster. We need a defer with switching to source cluster

Reply:
This will end up in failing all the subsequent tests in the pipeline

Reply:
Done